### PR TITLE
Add Integrations Test for Solvers

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

This PR:
- Introduces integration test for solvers
- Splits `api_url` in `user_api_url` and `management_api_url`
- Fixes: https://github.com/thoth-station/integration-tests/issues/8
- Refactor previous integration-tests
- split index and package metadata to facilitate interpretations of tests for each statement

### CODE REVIEW

File: features/solvers.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Feature: Check for solvers available in Thoth
+    Scenario: Check there are minimum solvers available in Thoth
+        Given a set of solvers
+            | name |
```
Comment (bissenbay) [inline features/solvers.feature:4]: @pacospace can we use `solver_name` to make it similar to the response body:
```json
{
  "parameters": {},
  "solvers": {
    "python": [
      {
        "os_name": "rhel",
        "os_version": "8",
        "python_version": "3.6",
        "solver_name": "solver-rhel-8-py36"
      },
      {
        "os_name": "fedora",
        "os_version": "31",
        "python_version": "3.8",
        "solver_name": "solver-fedora-31-py38"
      },
      {
        "os_name": "fedora",
        "os_version": "31",
        "python_version": "3.7",
        "solver_name": "solver-fedora-31-py37"
      }
    ]
  }
}
```

Comment (pacospace) [reply features/solvers.feature:4]: I totally agree, thanks @bissenbay

**Comment by pacospace** [comment]
Not sure why test is failing: @goern 

```
Feature: Check for solvers available in Thoth # features/solvers.feature:1

  Scenario: Check there are minimum solvers available in Thoth  # features/solvers.feature:2
    Given a set of solvers                                      # features/steps/solvers.py:27 0.000s
      | solver_name           |
      | solver-rhel-8-py36    |
      | solver-fedora-31-py37 |
      | solver-fedora-31-py38 |
    And deployment is accessible using HTTPS                    # features/steps/basic.py:31 1.762s
    When we ask if the solver is available                      # features/steps/solvers.py:36 0.694s
    Then we should find the solver available                    # features/steps/solvers.py:45 0.003s
      Assertion Failed: 
      Expected: a sequence over [<['solver-rhel-8-py36', 'solver-fedora-31-py37', 'solver-fedora-31-py38']>] in any order
           but: not matched: 'solver-rhel-8-py36'
      
      Captured stdout:
      {'requested_solvers': ['solver-rhel-8-py36', 'solver-fedora-31-py37', 'solver-fedora-31-py38'], 'available_solvers': ['solver-rhel-8-py36', 'solver-fedora-31-py38', 'solver-fedora-31-py37']}

Failing scenarios:
  features/solvers.feature:2  Check there are minimum solvers available in Thoth

0 features passed, 1 failed, 0 skipped
0 scenarios passed, 1 failed, 0 skipped
3 steps passed, 1 failed, 0 skipped, 0 undefined
Took 0m2.459s
```

File: features/steps/basic.py [hunk 2]
```diff
@@ -34,16 +34,26 @@ def deployment_accessible(context, scheme):
...
     ), f"Invalid response whn accessing User API /api/v1 endpoint: {response.status_code!r}"
 
     assert response.text, "Empty response from server for User API /api/v1 endpoint"
 
+    context.management_api_url = os.environ["THOTH_MANAGEMENT_API_URL"]
+    response = requests.get(f"{context.scheme}://{context.management_api_url}/api/v1", verify=False)
+
+    assert (
+        response.status_code == 200
+    ), f"Invalid response whn accessing Management API /api/v1 endpoint: {response.status_code!r}"
```
Comment (bissenbay) [inline features/steps/basic.py:53]: `when`

File: features/solvers.feature [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Feature: Check for solvers available in Thoth
+    Scenario: Check there are minimum solvers available in Thoth
+        Given a set of solvers
```
Comment (goern) [inline features/solvers.feature:3]: This feels strange, we are stating preconditions that we check for?! Shouldnt we simply start with the deployment being accessible and then we should get the list of solvers?

Comment (pacospace) [reply features/solvers.feature:3]:
```
Feature: Check for solvers available in Thoth
    Scenario: Check there are minimum solvers available in Thoth
        Given a set of solvers
            | solver_name |
            | solver-rhel-8-py36 |
            | solver-fedora-31-py37 |
            | solver-fedora-31-py38 |
        And deployment is accessible using HTTPS
        When we ask for the available solvers
        Then they should include at least the given solvers
```

The check for the deployment accessible comes right after, the first statement is a table which cannot fail. I can invert it :)

Comment (goern) [reply features/solvers.feature:3]: I think we shoulds assume/have the first table, we simply query, and the results should match the table

Comment (pacospace) [reply features/solvers.feature:3]: What about now?

```
Feature: Check for minimum solvers available in Thoth
    Scenario: Check there are minimum solvers available in Thoth
        Given deployment is accessible using HTTPS
        And a minimum set of solvers requested
            | solver_name |
            | solver-rhel-8-py36 |
            | solver-fedora-31-py37 |
            | solver-fedora-31-py38 |
        When we ask for the available solvers
        Then they should include at least the minimum set of solvers
```

**Comment by pacospace** [comment]
> Not sure why test is failing: @goern
> 
> ```
> Feature: Check for solvers available in Thoth # features/solvers.feature:1
> 
>   Scenario: Check there are minimum solvers available in Thoth  # features/solvers.feature:2
>     Given a set of solvers                                      # features/steps/solvers.py:27 0.000s
>       | solver_name           |
>       | solver-rhel-8-py36    |
>       | solver-fedora-31-py37 |
>       | solver-fedora-31-py38 |
>     And deployment is accessible using HTTPS                    # features/steps/basic.py:31 1.762s
>     When we ask if the solver is available                      # features/steps/solvers.py:36 0.694s
>     Then we should find the solver available                    # features/steps/solvers.py:45 0.003s
>       Assertion Failed: 
>       Expected: a sequence over [<['solver-rhel-8-py36', 'solver-fedora-31-py37', 'solver-fedora-31-py38']>] in any order
>            but: not matched: 'solver-rhel-8-py36'
>       
>       Captured stdout:
>       {'requested_solvers': ['solver-rhel-8-py36', 'solver-fedora-31-py37', 'solver-fedora-31-py38'], 'available_solvers': ['solver-rhel-8-py36', 'solver-fedora-31-py38', 'solver-fedora-31-py37']}
> 
> Failing scenarios:
>   features/solvers.feature:2  Check there are minimum solvers available in Thoth
> 
> 0 features passed, 1 failed, 0 skipped
> 0 scenarios passed, 1 failed, 0 skipped
> 3 steps passed, 1 failed, 0 skipped, 0 undefined
> Took 0m2.459s
> ```

Finally adjusted
```
Feature: Check for minimum solvers available in Thoth # features/solvers.feature:1

  Scenario: Check there are minimum solvers available in Thoth   # features/solvers.feature:2
    Given deployment is accessible using HTTPS                   # features/steps/basic.py:31 1.813s
    And a minimum set of solvers requested                       # features/steps/solvers.py:27 0.001s
      | solver_name           |
      | solver-rhel-8-py36    |
      | solver-fedora-31-py37 |
      | solver-fedora-31-py38 |
    When we ask for the available solvers                        # features/steps/solvers.py:37 0.672s
    Then they should include at least the minimum set of solvers # features/steps/solvers.py:46 0.002s

1 feature passed, 0 failed, 0 skipped
1 scenario passed, 0 failed, 0 skipped
4 steps passed, 0 failed, 0 skipped, 0 undefined
Took 0m2.488s
```

**Comment by goern** [comment]
and what about adding a `.thoth.yaml` ?

**Comment by pacospace** [comment]
> LGTM, could you rename `_url` to `_host` ? :)

Updated

**Comment by pacospace** [comment]
> and what about adding a `.thoth.yaml` ?

Added

**Pull request merged**